test/catch2/live_restore/api/test_live_restore_fh_lock_close_sync.cpp

```diff
@@ -0,0 +1,51 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+/*
+ * [live_restore_fh_lock_close_sync]: live_restore_fs.c
+ * Test some minimal file handle APIs.
+ */
+
+#include "../utils_live_restore.h"
+
+using namespace utils;
+
+TEST_CASE(
+  "Live Restore fh_lock fh_close fh_sync", "[live_restore],[live_restore_fh_lock_close_sync]")
+{
+    live_restore_test_env env;
+
+    std::string file_name = "file";
+    WT_FILE_HANDLE *fh = nullptr;
+    create_file(env.dest_file_path(file_name));
+    create_file(env.source_file_path(file_name));
+
+    WT_SESSION *wt_session = reinterpret_cast<WT_SESSION *>(env.session);
+    REQUIRE(env.lr_fs->iface.fs_open_file((WT_FILE_SYSTEM *)env.lr_fs, wt_session,
+              env.dest_file_path(file_name).c_str(), WT_FS_OPEN_FILE_TYPE_DATA, 0, &fh) == 0);
+
+    // If we test this any deeper then we'd be testing the posix implementation.
+    REQUIRE(fh->fh_lock(fh, wt_session, true) == 0);
+    REQUIRE(fh->fh_lock(fh, wt_session, false) == 0);
```

Comment:
I know we're testing the backing implementation, but we could have some basic tests for the API contract:
```suggestion
    REQUIRE(fh->fh_lock(fh, wt_session, true) == 0);
    
    // Re-entrancy is allowed/not allowed
    REQUIRE(fh->fh_lock(fh, wt_session, true) == XXXX);

    // A second session can't take the lock
    REQUIRE(fh->fh_lock(fh, wt_session2, true) == EBUSY);
    
    // A second session can't free the lock
    REQUIRE(fh->fh_lock(fh, wt_session2, false) == EBUSY);
    
    REQUIRE(fh->fh_lock(fh, wt_session, false) == 0);
    
    // A second session can now take the lock
    REQUIRE(fh->fh_lock(fh, wt_session2, true) == 0);
    REQUIRE(fh->fh_lock(fh, wt_session2, false) == EBUSY);
```

Reply:
File handle locks are per process to the majority of this is incorrect, barring the re-entrancy which is allowed.

Reply:
257e1725d5